201-front-door-waf-http-params/azuredeploy.json

```diff
@@ -34,23 +45,27 @@
         }
     },
     "variables": {
-        "frontdoorref": "[resourceId('Microsoft.Network/frontDoors', parameters('frontDoorName'))]",
-        "wafpolicyref": "[resourceId('Microsoft.Network/FrontDoorWebApplicationFirewallPolicies', parameters('wafPolicyName'))]",
-        "frontdoorLocation": "global"
+        "frontdoorLocation": "global",
+        "frontdoorSuffix": ".azurefd.net"
```

Comment:
use reference()

Reply:
The resource cannot make a reference to itself while is being created.